Add audioSource property to WaveConfig

vivekkumar056 opened the pull request:
This update introduces a new optional parameter in WaveConfig to let clients specify
a custom MediaRecorder.AudioSource. If not provided, the default source is used.
This enhances flexibility for various recording scenarios, such as voice communication
or recognition.

vivekkumar056 committed:
This update introduces a new optional parameter in WaveConfig to let clients specify
a custom MediaRecorder.AudioSource. If not provided, the default source is used.
This enhances flexibility for various recording scenarios, such as voice communication
or recognition.

squti commented on android-wave-recorder/src/main/java/com/github/squti/androidwaverecorder/WaveConfig.kt:
```diff
@@ -50,4 +55,29 @@ internal fun channelCount(channels: Int) = when (channels) {
     else -> throw IllegalArgumentException("Unsupported audio channel")
+}
+
+internal fun validateAudioSource(audioSource: Int) {
```
Could you explain why we need this validation function? I think the client is responsible for validating the supported audio source.

vivekkumar056 replied:
We need the validation function because relying only on the client is risky—clients can be misconfigured or send unsupported audio sources. Server-side validation ensures data integrity, provides clear error messages, and keeps the system stable. It also centralizes control, making it easier to manage and scale.

squti replied:
Don't use AI to answer my comment. Remove the validation function. Also fix your commit message. It should follow git convention.

vivekkumar056 replied:
I did not use AI to answer your comment. i have just rephrased what i wrote by my own, that's why it feels like that.
Anyway i'll remove and fix the commit message in next commit.

squti replied:
The commit message is too long. Please use this one:
Add audioSource property to WaveConfig

vivekkumar056 committed:
feat(waveconfig): add optional parameter to set custom AudioSource

Allows clients to specify a custom MediaRecorder.AudioSource in WaveConfig.
Defaults to the system source if not provided.
